### PR TITLE
First pass at changing .length to .span

### PR DESCRIPTION
**Pull request by hyanwong**
Closes https://github.com/tskit-dev/tskit/issues/29

### CODE REVIEW

**Comment by jeromekelleher** [comment]
Looks good, needs some tests though. Just need to ensure that length == span == (left - right) somewhere.

**Comment by hyanwong** [comment]
Done, shall I squash?

File: python/tskit/trees.py [hunk 1]
```diff
@@ -604,9 +604,8 @@ def branch_length(self, u):
 
         >>> tree.time(tree.parent(u)) - tree.time(u)
 
-        Note that this is not related to the value returned by
-        :attr:`.length`, which describes the length of the interval
-        covered by the tree in genomic coordinates.
+        (note that this is not related to the deprecated property :attr:`.length`
+        which is a deprecated alias for the genomic :attr:`.span` covered by a tree)
```
Comment (jeromekelleher) [inline python/tskit/trees.py]: deprecated twice here.

File: python/tests/test_highlevel.py [hunk 2]
```diff
@@ -1407,6 +1407,9 @@ def test_deprecated_apis(self):
         self.assertEqual(
             ts.get_pairwise_diversity(samples), ts.pairwise_diversity(samples))
         self.assertTrue(np.array_equal(ts.get_samples(), ts.samples()))
+        # Test APIs for a single tree in a ts
+        self.assertEqual(ts.first().get_length, ts.first().span)
+        self.assertEqual(ts.first().length, ts.first().span)
```
Comment (jeromekelleher) [inline python/tests/test_highlevel.py]: Be nice to check this somewhere on on not just the first tree. Also, check 
```
l, r = tree.interval
tree.span == r - 
```

Comment (hyanwong) [reply python/tests/test_highlevel.py]: Yep - I've just re-written it (and put it in a more sensible place). Thanks.

**Comment by jeromekelleher** [comment]
OK looks good, can you squash please? (The ``git commit --amend`` option is useful for this sort of thing...)

**Comment by hyanwong** [comment]
Done. Thanks for the tip. Will look into it

**Comment by jeromekelleher** [comment]
Thanks @hyanwong, merging.

**Pull request merged**